### PR TITLE
[Rocketmq-277] fix get localhost throw java.net.UnknownHostException,when server hostname  not in /etc/hosts

### PR DESCRIPTION
**Pull request by fuyou001**
Motivation:
 fix bug [ROCKETMQ-277](https://issues.apache.org/jira/browse/ROCKETMQ-277)

### CODE REVIEW

File: common/src/main/java/org/apache/rocketmq/common/MixAll.java [hunk 1]
```diff
@@ -407,12 +408,50 @@ private static String localhost() {
...
         }
     }
 
+    private static String getLocalhostByNetworkInterface() throws SocketException {
+        List<String> candidatesHost = new ArrayList<String>();
+        Enumeration<NetworkInterface> enumeration = NetworkInterface.getNetworkInterfaces();
+
+        while (enumeration.hasMoreElements()) {
+            NetworkInterface networkInterface = enumeration.nextElement();
+            if ("docker0".equals(networkInterface.getName()) || !networkInterface.isUp()) {
```
Comment (zhouxinyu) [inline common/src/main/java/org/apache/rocketmq/common/MixAll.java]: Do we have a test in docker container for this method ?

**Comment by vongosling** [comment]
Thank you everyone. I will make some polish and merge this PR :-)

File: common/src/main/java/org/apache/rocketmq/common/MixAll.java [hunk 2]
```diff
@@ -407,12 +408,50 @@ private static String localhost() {
...
+            } catch (Exception ignored) {
+            }
+
             throw new RuntimeException("InetAddress java.net.InetAddress.getLocalHost() throws UnknownHostException"
                 + FAQUrl.suggestTodo(FAQUrl.UNKNOWN_HOST_EXCEPTION),
                 e);
         }
     }
 
+    private static String getLocalhostByNetworkInterface() throws SocketException {
```
Comment (lizhanhui) [inline common/src/main/java/org/apache/rocketmq/common/MixAll.java]: RemotingUtil has already had a similar method, aka, getLocalAddress, is there any rational reason to create a duplicated one?

Comment (vongosling) [reply common/src/main/java/org/apache/rocketmq/common/MixAll.java]: I have marked it using FIXME label. reverse logic comparing to RemotingUtil method, we could consider to refactor it in RocketMQ 5.0.

**Pull request closed without merging**